platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+/**
+ * The migration context is defined as a base interface for all migration contexts.
```

Comment:
❓ Given the number of interfaces we have in this API bundle, do you think we should consider splitting them up between those to be used vs. the ones the framework expect to be implemented? Maybe have different sub-packages or notes in the comments?

Reply:
From an OSGi best practices standpoint, we really should split them up. Otherwise, to get the `Migratable` interface, the integrator is going to need to depend on a bunch of stuff they don't need, and then manually tweak the packaging of their bundle to cut down on bloat.